### ISSUE TITLE
Writing nested slice roundtrip fails (parquet)

### ISSUE THREAD

**Issue by ritchie46** [jorgecarleitao/arrow2#1394]
I tried to update to 7be484eb530805696c1ab79aae809f973d92cd5a, but this test failed in polars. 

```python
 def test_nested_sliced() -> None:
    for df in [
        pl.Series([[1, 2], [3, 4], [5, 6]]).slice(2, 2).to_frame(),
        pl.Series([[None, 2], [3, 4], [5, 6]]).to_frame(),
        pl.Series([[None, 2], [3, 4], [5, 6]]).slice(2, 2).to_frame(),
        pl.Series([["a", "a"], ["", "a"], ["c", "de"]]).slice(3, 2).to_frame(),
        pl.Series([[None, True], [False, False], [True, True]]).slice(2, 2).to_frame(),
    ]:
        f = io.BytesIO()
        df.write_parquet(f)
        f.seek(0)
       assert_frame_equal(pl.read_parquet(f), df)
```

```

left = shape: (1, 1)
┌───────────┐
│           │
│ ---       │
│ list[i64] │
╞═══════════╡
│ [null, 5] │
└───────────┘
right = shape: (1, 1)
┌───────────┐
│           │
│ ---       │
│ list[i64] │
╞═══════════╡
│ [5, 6]    │
└───────────┘

```

**Comment by jorgecarleitao** [comment]
On it

**Comment by jorgecarleitao** [comment]
Fixed by #1397
